common/src/sql/dbinit.sql

```diff
@@ -400,7 +400,9 @@ CREATE TABLE omicron.public.disk (
 
     /* Disk configuration */
     size_bytes INT NOT NULL,
-    origin_snapshot UUID
+    block_size INT NOT NULL,
```

Comment:
It's an interesting question here about what we want to store in the database. In general today we only have block sizes like 512 and 4096; however, there could be others that we want to logically advertise especially as SSDs continue to increase their internal block size or we want to give VMs more flexibility here. While an integer is the simplest, maybe it's worth considering a semantic enum or otherwise figuring out how we ensure that invalid and unsupported values don't make it in. I'm not sure what our general db strategy is in that respect.

Reply:
I'm also not sure what our general db strategy is here.

CRDB supports constraints, something like:

```
block_size INT NOT NULL CONSTRAINT block_size CHECK (block_size = 512 OR block_size = 4096),
```

Enums are more Rust work but possible too.

These check constraints can be changed using ALTER TABLE in the future if we plan on supporting more block sizes. If we only ever *add* supported block sizes, then that means adding another `OR block_size = X` clause:

```
root@:26257/omicron> alter table disk drop constraint block_size;
ALTER TABLE

Time: 91ms total (execution 91ms / network 0ms)

root@:26257/omicron> alter table disk add constraint block_size check (block_size = 512 OR block_size = 4096 OR block_size = 2048);
ALTER TABLE

Time: 102ms total (execution 101ms / network 0ms)

```

Rolling back that schema change is problematic because rows could have made it in with block size X, however [the documentation mentions that](https://www.cockroachlabs.com/docs/stable/check.html):

> The CHECK [constraint](https://www.cockroachlabs.com/docs/v21.2/constraints) specifies that values for the column in [INSERT](https://www.cockroachlabs.com/docs/v21.2/insert) or [UPDATE](https://www.cockroachlabs.com/docs/v21.2/update) statements must return TRUE or NULL for a Boolean expression.

so at least SELECTs will still work.

There would also have to be an additional change to Crucible to support any new block sizes.

Reply:
My inclination would be to use an enum.  We've done that in a bunch of places.  I don't think we use `check` much today.  `check` may well be just as good but on the surface it seems like an enum (with a corresponding Rust enum) would ensure only valid values could be represented in Rust as well as in SQL, while with `check` I think you could still easily construct invalid Rust representations.

Reply:
Added an enum in a20f0eb2b5d577c14cb599c6f48c329e8e350bc9